runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java

```diff
@@ -64,10 +72,228 @@ public void startReporter() {
                 .convertDurationsTo(TimeUnit.MILLISECONDS)
                 .build();
 
+        collectorRegistry = new CollectorRegistry();
+
         try {
             URL target = new URL(serverURL);
-            collectorRegistry = new CollectorRegistry();
-            collectorRegistry.register(new DropwizardExports(metricRegistry));
+
+//          Configuration for metrics without functions.
+            MapperConfig queryMetricConfig = new MapperConfig();
+            queryMetricConfig.setMatch("io.siddhi.SiddhiApps.*.Siddhi.Queries.*.*");
+            queryMetricConfig.setName("siddhi.query" + "." + "${2}");
+            Map<String, String> queryMetricLabels = new HashMap<String, String>();
+            queryMetricLabels.put("app", "${0}");
+            queryMetricLabels.put("type", "query");
+            queryMetricLabels.put("element", "${1}");
+            queryMetricLabels.put("metrics", "${2}");
+            queryMetricConfig.setLabels(queryMetricLabels);
+
+            MapperConfig sourceMapperMetricConfig = new MapperConfig();
+            sourceMapperMetricConfig.setMatch("io.siddhi.SiddhiApps.*.Siddhi.SourceMappers.*.*");
+            sourceMapperMetricConfig.setName("siddhi.sourcemapper" + "." + "${2}");
+            Map<String, String> sourceMapperMetriclabels = new HashMap<String, String>();
+            sourceMapperMetriclabels.put("app", "${0}");
+            sourceMapperMetriclabels.put("type", "sourcemapper");
+            sourceMapperMetriclabels.put("element", "${1}");
+            sourceMapperMetriclabels.put("metrics", "${2}");
+            sourceMapperMetricConfig.setLabels(sourceMapperMetriclabels);
+
+            MapperConfig sinkMapperMetricConfig = new MapperConfig();
+            sinkMapperMetricConfig.setMatch("io.siddhi.SiddhiApps.*.Siddhi.SinkMappers.*.*");
+            sinkMapperMetricConfig.setName("siddhi.sikmapper" + "." + "${2}");
```

Comment:
Why are we contacting constant strings

Reply:
The above requested change is addressed by the commit 2523d00.